test/unit/configuration.test.ts

```diff
@@ -41,153 +44,296 @@ describe('Sequelize', () => {
 
-      expect(options.dialect).to.equal('mysql');
+      expect(options.dialect).to.equal(dialect);
 
       expect(config.database).to.equal('dbname');
       expect(config.host).to.equal('example.com');
       expect(config.username).to.equal('user');
       expect(config.password).to.equal('pass');
-      expect(config.port).to.equal('9821');
-    });
-
-    describe('SQLite path inititalization', () => {
-      if (dialect === 'sqlite') {
-        it('should accept relative paths for sqlite', () => {
-          const sequelize = new Sequelize('sqlite:subfolder/dbname.db');
-          const options = sequelize.options;
-          expect(options.dialect).to.equal('sqlite');
-          expect(options.storage).to.equal(path.resolve('subfolder', 'dbname.db'));
-        });
-
-        it('should accept absolute paths for sqlite', () => {
-          const sequelize = new Sequelize('sqlite:/home/abs/dbname.db');
-          const options = sequelize.options;
-          expect(options.dialect).to.equal('sqlite');
-          expect(options.storage).to.equal(path.resolve('/home/abs/dbname.db'));
-        });
-
-        it('should prefer storage in options object', () => {
-          const sequelize = new Sequelize('sqlite:/home/abs/dbname.db', { storage: '/completely/different/path.db' });
-          const options = sequelize.options;
-          expect(options.dialect).to.equal('sqlite');
-          expect(options.storage).to.equal(path.resolve('/completely/different/path.db'));
-        });
-
-        it('should be able to use :memory:', () => {
-          const sequelize = new Sequelize('sqlite://:memory:');
-          const options = sequelize.options;
-          expect(options.dialect).to.equal('sqlite');
-
-          // empty host is treated as :memory:
-          expect(options.host).to.equal('');
-          expect(options.storage).to.equal(undefined);
-        });
-      }
+      expect(config.port).to.equal(9821);
+      expect(config.replication.write).to.deep.eq({
+        database: 'dbname',
+        username: 'user',
+        password: 'pass',
+        host: 'example.com',
+        port: 9821,
+        protocol: 'tcp',
+        ssl: undefined,
+        dialectOptions: {},
+      });
+      expect(config.replication.read).to.deep.eq([]);
     });
 
-    it('should work with no authentication options', () => {
-      const sequelize = new Sequelize('mysql://example.com:9821/dbname');
+    it('supports not providing username, password, or port', () => {
+      const sequelize = new Sequelize(`${dialect}://example.com/dbname`);
       const config = sequelize.config;
 
-      expect(config.username).to.not.be.ok;
-      expect(config.password).to.be.null;
-    });
-
-    it('should work with no authentication options and passing additional options', () => {
-      const sequelize = new Sequelize('mysql://example.com:9821/dbname', {});
-      const config = sequelize.config;
+      const defaultPort: Record<Dialect, number> = {
```

Comment:
Don't you want to use the getter for this?

Reply:
`dialect.getDefaultPort()`? No because it's an opportunity to test it

Reply:
Yeah, ignore this. I was a bit confused before but this makes sense